### ISSUE TITLE
Using once(TOPIC) multiple times on the same topic, may lead to a miss call of onadd.

### ISSUE THREAD

**Issue by jeromevalentin** [RangerMauve/mqtt-emitter#4]
Using once(TOPIC) multiple times on the same topic, may lead to a miss call of onadd.
The problem is easily reproductible, (See the case below)

``` javascript
import MqttEmitter from "mqtt-emitter";

class MyEmitter extends MqttEmitter {
  constructor() {
    super();
  }
  onadd(topic) {
    console.log("onadd",topic);
  }
  onremove(topic) {
    console.log("onremove",topic);
  }
}

let emitter = new MyEmitter();

emitter.once("A/B", (msg) => {
  console.log("1 =>",msg);
});
emitter.emit("A/B", "foo");

emitter.once("A/B", (msg) => {
  console.log("2 =>",msg);
});
emitter.emit("A/B", "bar");
```

The console output will be:

`onadd A/B`
`1 => foo`
`onremove A/B`
`2 => bar`
`onremove A/B`

while we expect to have:

`onadd A/B`
`1 => foo`
`onremove A/B`
`onadd A/B`
`2 => bar`
`onremove A/B`

When binding mqtt-emitter with a real mqtt implementation, this leads to a lost of message!

Regards,
Jérôme

**Comment by jeromevalentin** [comment]
The problem is coming from the test that is causing the call to onadd in addListener (the library call onadd only when the store has undefined listeners).
Unfortunately, the removeListener method does not clean up the store, so on the next addListener call, the store contains an empty listener array which is no more undefined.

A simple fix consists to upgrade the addListener method like below:

``` javascript
function addListener(topic, handler) {
    var matcher = mqtt_regex(topic);
    var topic_string = matcher.topic;

    var listeners = this._listeners.get(topic_string);
    if (!listeners) {
        listeners = this._listeners.set(topic_string, []);
    }
// FIX HERE BY CALLING ONADD AS SOON AS THERE IS NO LISTENER REGISTERED
    var is_new = (listeners.length == 0);

    listeners.push({
        fn: handler,
        params: matcher.exec,
        pattern: topic
    });

    if (is_new) this.onadd(topic_string);

    return this;
}
```

Hope it helps,
Regards,
Jérôme

**Comment by RangerMauve** [comment]
Sorry, I'm on vacation for the week, I'll hopefully get to this this weekend.

**Comment by RangerMauve** [comment]
Thanks for the bug report! Fixed in 1.2.4
